packages/react-components/src/components/ChatMessage/ChatMessageContent.tsx

```diff
@@ -84,12 +83,74 @@ const MessageContentAsRichTextHTML = (props: ChatMessageContentProps): JSX.Eleme
     });
   }, [props]);
 
+  const processNodeDefinitions = ProcessNodeDefinitions();
```

Comment:
Same here, extract this part into a util function so no need to create it for each single render